receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -319,3 +334,60 @@ func (mf *metricFamily) appendMetric(metrics pmetric.MetricSlice) {
 
 	metric.MoveTo(metrics.AppendEmpty())
 }
+
+func (mf *metricFamily) addExemplar(l labels.Labels, e exemplar.Exemplar) error {
+	gk := mf.getGroupKey(l)
+	mg := mf.groups[gk]
+	if mg == nil {
+		return nil
+	}
+	es := mg.exemplars
+	err := convertToOtlpExemplar(e, es.AppendEmpty())
+	return err
```

Comment:
this has a limitation that exemplars parsed before the data/value timeseries are ignored. Is that acceptable? If yes, we should document in the caveats section of the exemplar support.

Reply:
Done, adding traceid and spanid only if err == nil from hex decode